packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port_utils.py

```diff
@@ -0,0 +1,106 @@
+import logging
+import shutil
+from pathlib import Path
+from typing import Any, Callable, Coroutine, Dict, Optional
+
+from ..node_ports import config, data_items_utils, filemanager
+from .links import DownloadLink, FileLink, ItemConcreteValue, PortLink
+
+log = logging.getLogger(__name__)
+
+
+async def get_value_from_link(
+    key: str,
+    value: PortLink,
+    fileToKeyMap: Optional[Dict[str, str]],
+    node_port_creator: Callable[[str], Coroutine[Any, Any, Any]],
+) -> ItemConcreteValue:
+    log.debug("Getting value %s", value)
+    # create a node ports for the other node
+    other_nodeports = await node_port_creator(value.node_uuid)
+    # get the port value through that guy
+    log.debug("Received node from DB %s, now returning value", other_nodeports)
+
+    value = await other_nodeports.get(value.output)
+    if isinstance(value, Path):
+        file_name = value.name
+        # move the file to the right final location
+        # if a file alias is present use it
+        if fileToKeyMap:
+            file_name = next(iter(fileToKeyMap))
+
+        file_path = data_items_utils.create_file_path(key, file_name)
+        if value == file_path:
+            # this can happen in case
```

Comment:
in case ... what? :-)

Reply:
good catch. that was a copy paste from node_ports_v1.. which I will correct as well.

Reply:
ok so I cannot change in the node_ports_v1 cause that is the very file that makes black hook fail with error 123... and I don't know how to make black skip it but by changing the pre-commit-config.yml file.